app/src/main/java/tech/relaycorp/letro/repository/GatewayRepository.kt

```diff
@@ -62,6 +83,187 @@ class GatewayRepository @Inject constructor(
+        thirdPartyEndpointNodeId: String?,
+        authorizedReceivingMessagesFromServer: Boolean?,
+    ) = authorizedReceivingMessagesFromServer != true &&
+        firstPartyEndpointNodeId != null &&
+        thirdPartyEndpointNodeId != null
+
+    private fun generatePairingRequest(
+        requesterVeraId: String,
+        contactVeraId: String,
+        requesterEndpoint: FirstPartyEndpoint,
+    ): ByteArray {
+        val publicKey = requesterEndpoint.publicKey
+        val publicKeyBase64 = Base64.encodeToString(publicKey.encoded, Base64.NO_WRAP)
+        val content = "$requesterVeraId,$contactVeraId,$publicKeyBase64"
+        return content.toByteArray()
+    }
+
+    private fun parsePairingMatch(content: ByteArray): PairingMatchDataModel {
+        val contentString = content.toString(Charset.defaultCharset())
+        val parts = contentString.split(",")
+        return PairingMatchDataModel(
+            senderVeraId = parts[0],
+            receiverVeraId = parts[1],
+            receiverEndpointId = parts[2],
+            receiverEndpointPublicKey = Base64.decode(parts[3], Base64.NO_WRAP),
+        )
+    }
+
+    private suspend fun generatePairingAuth(
+        match: PairingMatchDataModel,
+        firstPartyEndpoint: FirstPartyEndpoint,
+    ): ByteArray {
+        // TODO DELETE THIS COMMENTED CODE
+        // NOTE to Gus: All this logic is done before in the AccountRepository
+//        // Implement some app-specific logic to check that the pairing request exists.
+//        if (!contactRequestExists(match.requesterVeraId, match.contactVeraId)) {
+//            // Granting authorisation is a sensitive operation and we shouldn't blindly
+//            // trust the server.
+//            throw PairingRequestException("Pairing request does not exist ($match)")
+//        }
+//
+//        // Implement some app-specific logic to store the contact's Awala endpoint id, as
+//        // we'll need it later to (a) complete pairing and (b) send messages to them.
+//        storeContactAwalaId(
+//            match.requesterVeraId,
+//            match.contactVeraId,
+//            match.contactEndpointId,
+//        )
+
+        return firstPartyEndpoint.authorizeIndefinitely(
+            match.receiverEndpointPublicKey,
+        )
+    }
+
+    // TODO Potentially delete the following function
+    // NOTE to Gus: In this implementation this function is not needed.
+    // Only if you tell me that the import has to be done for Gateway to function properly
+//    private suspend fun importPairingAuth(auth: ByteArray) {
+//        // For the following line, I'm getting Cannot access class 'tech.relaycorp.relaynet.SessionKey'. Check your module classpath for missing or conflicting dependencies
+//        val contactEndpoint = PrivateThirdPartyEndpoint.import(auth)
+//
+//        // TODO DELETE THIS COMMENTED CODE
+//        // NOTE to Gus: All this logic is done after in the AccountRepository
```

Comment:
I couldn't find something equivalent in there. Do you really mean `AccountRepository`?

Reply:
@gnarea ContactRepository*

Reply:
Discussed and resolved